apk/src/com/android/healthconnect/controller/utils/DeviceInfoUtils.kt

```diff
@@ -54,6 +56,9 @@ class DeviceInfoUtilsImpl @Inject constructor() : DeviceInfoUtils {
                 if (!TextUtils.isEmpty(info.activityInfo.packageName)) {
                     try {
                         val ai = pm.getApplicationInfo(info.activityInfo.packageName, 0)
+                        if (GmsCompat.isGmsApp(ai)) {
+                            return FEEDBACK_REPORTER
```

Comment:
Should check first whether package name matches. isGmsApp() returns true for several GMS components, not only GmsCore (Play services).